src/Utils/messages.ts

```diff
@@ -406,6 +408,61 @@ export const generateWAMessageContent = async (
 				}
 			}
 		}
+	} else if ('stickerPack' in message) {
+		const { stickers, cover, name, publisher, packId, description } = message.stickerPack
+
+		const coverBuffer = await toBuffer((await getStream(cover)).stream)
+		const imageDataHash = sha256(coverBuffer).toString('base64')
+
+		const [coverUploadResult, ...stickerUploadResults] = await Promise.all([
+			prepareWAMessageMedia({ image: coverBuffer }, { ...options, mediaTypeOverride: 'image' }),
+			...stickers.map(s => prepareWAMessageMedia({ sticker: s.data }, { ...options, mediaTypeOverride: 'sticker' }))
+		])
+
+		const stickerPackIdValue = packId || generateMessageIDV2()
+		const coverImage = coverUploadResult.imageMessage!
+
+		const stickerPackSize = stickerUploadResults.reduce(
+			(acc, s) => acc + (s.stickerMessage?.fileLength ? +s.stickerMessage.fileLength : 0),
+			0
+		)
+
+		m.stickerPackMessage = {
+			name: name,
+			publisher: publisher,
+			stickerPackId: stickerPackIdValue,
+			packDescription: description,
+			stickerPackOrigin: WAProto.Message.StickerPackMessage.StickerPackOrigin.THIRD_PARTY,
+			stickerPackSize,
+
+			stickers: stickerUploadResults.map((uploadResult, i) => {
+				const stickerMsg = uploadResult.stickerMessage!
+				const fileSha256B64 = Buffer.from(stickerMsg.fileSha256!).toString('base64')
+
+				return {
+					fileName: `${fileSha256B64}.webp`,
+					mimetype: stickerMsg.mimetype,
+					isAnimated: stickerMsg.isAnimated,
+					emojis: stickers[i].emojis || [],
+					accessibilityLabel: stickers[i].accessibilityLabel
+				}
+			}),
```

Comment:
we need to make a zip file that has the stickers as images. They have weird file names but I forgot what they were for. In any case, the clients sending packs, do not outline it in this specific manner

Reply:
![image](https://github.com/user-attachments/assets/2d618df1-ca8f-45e4-853e-6ef236156fdf)

This is an image from March 6, 2025. The direct path led me to a zip file that I manually decrypted (it uses a new HKDF as well)

Reply:
im lacking the time and the immediate data to mention more things (this is all I have at the moment in my investigative report), but this should be an interesting start for you.

Reply:
the HKDF key was "WhatsApp Sticker Pack Keys"

Reply:
> Thank you guys, @purpshell and @SheIITear. This is very helpful. I will work on this in the next weekend.

Thanks for working on this! If it wasn't for Shelltear, I wouldn't have had an active sample. 

Make sure to credit :P, haha!